packages/application-shell/src/components/navbar/navbar.js

```diff
@@ -207,7 +208,7 @@ export class MenuItemLink extends React.PureComponent {
           if (this.props.useFullRedirectsForLinks) {
             event.preventDefault();
             this.redirectTo(this.props.linkTo);
-          }
+          } else this.props.onClick && this.props.onClick();
```

Comment:
We need to pass the `track` function as an argument, otherwise we can't use it "outside of the appshell".

```js
onClick(event, track)
```

To get the `track` function, you can access it from the context:

```js
static contextType = GtmContext;
```

Reply:
Why can't we `injectTracking` in application entry point instead? Passing `track` in click handler make it very tracking specific. In case we gonna do it, i'd rename `onMenuItemClick` -> `onTrackMenuItemClick`.

Reply:
It won't work, the AppShell is the top level component, so you can't wrap anything around that.

Reply:
Maybe we can avoid passing the `track` object and you can call push the tracking event directly:

```js
onMenuItemClick={event => {
  // extract info from event.currentTarget, then push it to the dataLayer
  window.dataLayer.push({
    event: 'TrackingEvent',
    trackingCategory: '',
    trackingAction: '',
    trackingLabel: '',
  });
}}
```

That's basically what the `track` function does